viadot/sources/s3.py

```diff
@@ -1,266 +1,165 @@
-                overwrite=overwrite,
-            )
-        except FileExistsError:
-            # Show a useful error message.
-            if recursive:
-                msg = f"At least one file in '{to_path}' already exists. Specify `overwrite=True` to overwrite."  # noqa
-            else:
-                msg = f"The file '{to_path}' already exists. Specify `overwrite=True` to overwrite."
-            raise FileExistsError(msg)
-
-        self.logger.info(
-            f"Successfully uploaded file(s) from '{from_path}' to '{to_path}'."
-        )
-
-    def exists(self, path: str = None) -> bool:
+    def exists(self, path: str) -> bool:
         """
-        Check if a location exists in Azure Data Lake.
+        Check if a location exists in S3.
 
         Args:
             path (str): The path to check. Can be a file or a directory.
-
-        Example:
-        ```python
-        from viadot.sources import AzureDataLake
-
-        lake = AzureDataLake(gen=1)
-        lake.exists("tests/test.csv")
-        ```
-
         Returns:
             bool: Whether the paths exists.
         """
-        path = path or self.path
-        return self.fs.exists(path)
+        return wr.s3.does_object_exist(boto3_session=self.session, path=path)
 
-    def download(
-        self,
-        to_path: str,
-        from_path: str = None,
-        recursive: bool = False,
-        overwrite: bool = True,
-    ) -> None:
+    def cp(self, from_path: str, to_path: str, recursive: bool = False):
+        """
+        Copies the contents of `from_path` to `to_path`.
 
-        if overwrite is False:
-            raise NotImplemented(
-                "Currently, only the default behavior (overwrite) is available."
-            )
+        Args:
+            from_path (str, optional): S3 Path for the source directory.
+            to_path (str, optional): S3 Path for the target directory.
+            recursive (bool): Set this to true if working with directories.
+                Defaults to False.
 
-        from_path = from_path or self.path
-        self.fs.download(rpath=from_path, lpath=to_path, recursive=recursive)
+        Example:
+            Copy files within two S3 locations:
 
-    @add_viadot_metadata_columns
-    def to_df(
-        self,
-        path: str = None,
-        sep: str = "\t",
-        quoting: int = 0,
-        lineterminator: str = None,
-        error_bad_lines: bool = None,
-    ):
-        if quoting is None:
-            quoting = 0
+            ```python
 
-        path = path or self.path
-        url = os.path.join(self.base_url, path)
+            from viadot.sources.s3 import S3
 
-        if url.endswith(".csv"):
-            df = pd.read_csv(
-                url,
-                storage_options=self.storage_options,
-                sep=sep,
-                quoting=quoting,
-                lineterminator=lineterminator,
-                error_bad_lines=error_bad_lines,
+            s3_session = S3()
```

Comment:
Would just call it `s3` since session is a concept not everyone might be familiar with.

Reply:
Updated as per [🎨 minor docstring change](https://github.com/dyvenia/viadot/pull/579/commits/fe28c1298f48fbda9d7d88e46c749521c115495f)